src/main/java/ch/njol/skript/expressions/base/PropertyExpression.java

```diff
@@ -30,13 +30,15 @@ public abstract class PropertyExpression<F, T> extends SimpleExpression<T> {
 	 *
 	 * @param property the name of property
 	 * @param fromType the types that the property should apply to
+	 * @param defaultExpression whether the types should be optional
 	 * @return an array of strings that represent the patterns of the provided property and types
 	 */
 	@SuppressWarnings("ConstantValue")
-	public static String[] getPatterns(String property, String fromType) {
+	public static String[] getPatterns(String property, String fromType, boolean defaultExpression) {
```

Comment:
You should overload the method and default `defaultExpression` to false.
Or better yet, remove the boolean parameter and make them two separate methods.